server/segment_rpc.go

```diff
@@ -254,6 +219,78 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 	w.Write(buf)
 }
 
+// Payment receives payment from Gateway and adds it into the orchestrator's balance
+func (h *lphttp) Payment(w http.ResponseWriter, r *http.Request) {
+	payment, segData, oInfo, ctx, err := h.processPaymentAndSegmentHeaders(w, r)
+	if err != nil {
+		return
+	}
+
```

Comment:
Double checking whether its OK to leave the body un-read here? If so, maybe leave a comment mentioning that is intentional?

Reply:
I think you don't need to explicitly close the body on  the server side.

Reply:
Yeah, just that it is unusual so a comment explaining that would be nice.

BTW while I agree this is ok from a golang resource-management perspective, I have seen HTTP clients get blocked if the server ignores the body. Probably not an issue here since there is no body to begin with, but again just highlighting it is kind of unusual to not drain the body.

Reply:
ok, I can add a comment (probably in some follow-up PR, since this one is merged).